### PR TITLE
node: fix genesis on start up

### PR DESCRIPTION
**Pull request by cmwaters**
After observing a few failed e2e tests I realised that I missed an edge case (when genesis has no validators) for this PR: https://github.com/tendermint/tendermint/pull/6554.

### CODE REVIEW

File: node/setup.go [hunk 1]
```diff
@@ -295,7 +295,7 @@ func createEvidenceReactor(
 
 	evidencePool, err := evidence.NewPool(logger, evidenceDB, sm.NewStore(stateDB), blockStore)
 	if err != nil {
-		return nil, nil, nil, err
+		return nil, nil, nil, fmt.Errorf("creating evidence pool: %v", err)
```
Comment (tychoish) [inline node/setup.go:298]: I think this should be a %w

**Review by tychoish — APPROVED** [review]
We currently have a genesis state store within the node, which seems quirky without having more context. Is there a path to being able to avoid this? https://github.com/tendermint/tendermint/blob/master/node/node.go#L75

How much of the e2e failures does this account for? Should it be backported to 0.34?

File: node/setup.go (hunk 1 above)
Comment (cmwaters) [reply node/setup.go:298]: correct

**Comment by cmwaters** [comment]
> We currently have a genesis state store within the node, which seems quirky without having more context. Is there a path to being able to avoid this?

We probably don't need this since we can derive it from the genesis doc

**Review by alexanderbez — APPROVED** [review]
